backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js

```diff
@@ -0,0 +1,388 @@
+    ].join(' ');
+    const cspObjectSrc = [
+        '\'none\'',
+    ].join(' ');
+    const cspFormActionSrc = [
+        '\'none\'',
+    ].join(' ');
+    const cspConnectSrc = [
+        '\'self\'',
+        dataApiUrl,
+        s3Url,
+        'https://cognito-idp.us-east-1.amazonaws.com'
+    ].join(' ');
+
+    return `${[
+        `default-src ${cspDefaultSrc};`,
+        `manifest-src ${cspManifestSrc};`,
+        `script-src ${cspScriptSrc};`,
+        `script-src-elem ${cspScriptSrcElem};`,
+        `script-src-attr ${cspScriptSrcAttr};`,
+        `worker-src ${cspWorkerSrc};`,
+        `style-src ${cspStyleSrc};`,
+        `style-src-elem ${cspStyleSrcElem};`,
+        `style-src-attr ${cspStyleSrcAttr};`,
+        `font-src ${cspFontSrc};`,
+        `img-src ${cspImgSrc};`,
+        `media-src ${cspMediaSrc};`,
+        `frame-src ${cspFrameSrc};`,
+        `frame-ancestors ${cspFrameAncestorsSrc};`,
+        `object-src ${cspObjectSrc};`,
+        `form-action ${cspFormActionSrc};`,
+        `connect-src ${cspConnectSrc};`,
+    ].join(' ')}`;
+};
+
+// ================================================================================================
+// =                                        TESTS                                                 =
+// ================================================================================================
+describe(testFilename(__filename), () => {
+    describe('Cloudfront security headers', () => {
+        it('should successfully return the security headers for csg prod', async () => {
+            const environment = environments.csg.prod;
+            const request = {
+                headers: {
+                    host: [{
+                        value: environment.webFrontend,
+                    }],
+                },
+            };
+            const response = {
+                headers: {},
+            };
+            const config = lambdaConfig({
+                lambdaPath: `index.js`,
+                request,
+                response,
+            });
+            const result = await runLambda(config);
+
+            // console.log(JSON.stringify(result, null, 2)); // @DEBUG
+
+            expect(result.headers['strict-transport-security'][0].key).to.equal('Strict-Transport-Security');
+            expect(result.headers['strict-transport-security'][0].value).to.equal('max-age=31536000; includeSubdomains; preload');
+            expect(result.headers['x-content-type-options'][0].key).to.equal('X-Content-Type-Options');
+            expect(result.headers['x-content-type-options'][0].value).to.equal('nosniff');
+            expect(result.headers['x-frame-options'][0].key).to.equal('X-Frame-Options');
+            expect(result.headers['x-frame-options'][0].value).to.equal('DENY');
+            expect(result.headers['x-xss-protection'][0].key).to.equal('X-Xss-Protection');
+            expect(result.headers['x-xss-protection'][0].value).to.equal('1; mode=block');
```

Comment:
This looks to be nonstandard. If you think we need it, mind adding a comment as to why, and perhaps point out why (or how) the vulnerability it describes is not an issue for us?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

Reply:
Removed.